lib/forms/Form.js

```diff
@@ -319,8 +319,16 @@ function MLForm$$createForm(schema, hostObject, formData, template) {
                     , modelPath = schema.modelPath;
 
                 if (component) {
-                    var parentEl = component.el.parentNode;
-                    parentEl.classList.toggle(FORM_VALIDATION_FAILED_CSS_CLASS, ! response.valid);
+                    // check component can render validation failure
+                    if (component.renderValidationFailure) {
+                        component.renderValidationFailure(schema);
```

Comment:
Might be worth passing `response` to this function too?  I assume it contains validation failure reasons?   If so, it gives components the ability to customise how they render errors further

Reply:
At the moment `renderValidationFailure` gets the schema of that particular component, which contains validation function and message inside, so as if the component needs it can use it. 

`response` contains information about the component as whole and error messages to show on error model popup. An example response below,

```
{  
   "valid":false,
   "reason":"one or more missing",
   "reasonCode":"REQUIRED",
   "path":".imagesCaptions"
}
```

Reply:
So the idea is if component wants validation info, get the validation function from schema and execute it. Is that make sense ?

Reply:
> So the idea is if component wants validation info, get the validation function from schema and execute it. Is that make sense ?

Would that result in the same output as what is stored in `response`?

Reply:
Yes, it is the same, but depends on input values passed in. For example, a component with multiple form inputs can run validation against each of its input values and find which one is causing the issue.

Reply:
Feels like redundant, but can pass `response` as another argument to function if needed.

Reply:
Hey, sorry - I forgot to reply!  So...

If my validation function checked that some value was:

- Not null
- A string less than 10 chars long

(Meaning the validation function could output two reasons for failure (null input OR <10 chars)... 

- Would `response` contain the reason (It looks like it would).

I don't understand how passing only the `schema` would allow me to work how why the validation failed.  Manually executing the schema's validation function seems somewhat unnecessary if we could just pass the `response` object through.

Of course...I may be grossly misunderstanding something! :)

Reply:
Actually dont know today I think would be useful :), updated.